gcc/rust/expand/rust-proc-macro.h

```diff
@@ -0,0 +1,190 @@
+// This file is part of GCC.
+
+// GCC is free software; you can redistribute it and/or modify it under
+// the terms of the GNU General Public License as published by the Free
+// Software Foundation; either version 3, or (at your option) any later
+// version.
+
+// GCC is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+// for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with GCC; see the file COPYING3.  If not see
+// <http://www.gnu.org/licenses/>.
+
+#ifndef RUST_PROC_MACRO_H
+#define RUST_PROC_MACRO_H
+
+#include <string>
+#include "rust-parse.h"
+#include "rust-proc-macro-invoc-lexer.h"
+#include "rust-hir-map.h"
+#include "rust-name-resolver.h"
+#include "rust-session-manager.h"
+#include "rust-ast.h"
+#include "rust-ast-collector.h"
+#include "rust-token-converter.h"
+#include "libproc_macro/proc_macro.h"
+
+namespace Rust {
+
+/**
+ * Load a procedural macro library and return a pointer to it's entrypoint.
+ *
+ * @param The path to the shared object file to load.
+ */
+const std::vector<ProcMacro::Procmacro>
+load_macros (std::string path);
+
+class ProcMacroExpander
+{
+public:
+  ProcMacroExpander (Session &session)
+    : session (session), has_changed_flag (false),
+      resolver (Resolver::Resolver::get ()),
+      mappings (Analysis::Mappings::get ())
+
+  {}
+
+  ~ProcMacroExpander () = default;
+
+  void import_proc_macros (std::string extern_crate);
```

Comment:
We'll have to see if this should be part of the EarlyNameResolver instead

Reply:
Will be done in an upcoming PR, reference issue is #2275